src/lint/lint.ts

```diff
@@ -87,3 +104,64 @@ export async function lint(options: Options, config: ProjectConfig) {
     return new CommandResult(1);
   }
 }
+
+async function fix(
+    warnings: FixableWarning[],
+    options: Options,
+    config: ProjectConfig,
+    analyzer: Analyzer,
+    analysis: Analysis) {
+  if (options.input) {
+    const packageRelativeInputFilenames = new Set(options.input.map(
+        (i) => path.relative(config.root, path.resolve('.', i))));
+    warnings = warnings.filter(
+        (w) => packageRelativeInputFilenames.has(w.sourceRange.file));
+  }
+
+  const fixes = warnings.map((w) => w.fix).filter((fix) => fix !== undefined) as
+      Replacement[][];
+  if (fixes.length === 0) {
+    console.log('No fixes to apply.');
+    return;
+  }
+
+  const {appliedEdits, incompatibleEdits, editedFiles} =
+      await applyEdits(fixes, makeParseLoader(analyzer, analysis));
+
+  for (const [newPath, newContents] of editedFiles) {
+    await fs.writeFile(
+        path.join(config.root, newPath), newContents, {encoding: 'utf8'});
+  }
+
+  const changeCountByFile = new Map<string, number>();
+  for (const appliedEdit of appliedEdits) {
+    for (const replacement of appliedEdit) {
+      changeCountByFile.set(
+          replacement.range.file,
+          (changeCountByFile.get(replacement.range.file) || 0) + 1);
+    }
+  }
+
+  for (const [file, count] of changeCountByFile) {
+    console.log(`  Made ${count} change${plural(count)} to ${file}`);
+  }
+
+  if (incompatibleEdits.length > 0) {
+    console.log(
+        `\nFixed ${appliedEdits.length} ` +
+        `warning${plural(appliedEdits.length)}, ` +
+        `${incompatibleEdits.length} had conflicts with other fixes. ` +
```

Comment:
As a user I should be told which files these occurred in, assuming that it is be possible for two fixes to conflict indefinitely (or for a fix to inadvertently cause a new error)